mesonbuild/backend/backends.py

```diff
@@ -455,10 +456,35 @@ def get_external_rpath_dirs(self, target):
                 args.extend(self.environment.coredata.get_external_link_args(target.for_machine, lang))
             except Exception:
                 pass
+        # Match rpath formats:
+        # -Wl,-rpath=
+        # -Wl,-rpath,
+        rpath_regex = re.compile(r'-Wl,-rpath[=,]([^,]+)')
+        # Match solaris style compat runpath formats:
+        # -Wl,-R
+        # -Wl,-R,
+        runpath_regex = re.compile(r'-Wl,-R[,]?([^,]+)')
+        # Match symbols formats:
+        # -Wl,--just-symbols=
+        # -Wl,--just-symbols,
+        symbols_regex = re.compile(r'-Wl,--just-symbols[=,]([^,]+)')
         for arg in args:
-            if arg.startswith('-Wl,-rpath='):
-                for dir in arg.replace('-Wl,-rpath=','').split(':'):
+            rpath_match = rpath_regex.match(arg)
+            if rpath_match:
+                for dir in rpath_match.group(1).split(':'):
                     dirs.add(dir)
+            runpath_match = runpath_regex.match(arg)
+            if runpath_match:
+                for dir in runpath_match.group(1).split(':'):
+                    # The symbols arg is an rpath if the path is a directory
+                    if Path(dir).is_dir():
+                        dirs.add(dir)
+            symbols_match = symbols_regex.match(arg)
+            if symbols_match:
+                for dir in symbols_match.group(1).split(':'):
+                    # Prevent usage of --just-symbols to specify rpath
+                    if Path(dir).is_dir():
+                        raise MesonException('Invalid arg for --just-symbols, {} is a directory.'.format(dir))
```

Comment:
Actually seems it's easy enough to blacklist `-Wl,--just-symbols` from being used to set an rpath.

Reply:
Mentioning -Wl,--just-symbols at all in this patch just clutters what should have been a tiny fix.

Better to consider the abuse of -Wl,--just-symbols an uncaught error.  I've never seen it misused in that way, have you?

Reply:
> Mentioning -Wl,--just-symbols at all in this patch just clutters what should have been a tiny fix.

I think we're a bit past that point since we also have to handle `-R`.

> Better to consider the abuse of -Wl,--just-symbols an uncaught error.

But it's not a build error in meson at all when used with the default gnu ld bfd linker, I don't want what should be a build time error turn into a runtime error. What I want to avoid is rpath being set using `-Wl,--just-symbols` and meson not recognizing the inconsistent state/behavior it is in. I much prefer meson to hard error if it can not proceed in a sane way as this greatly simplifies tracking down bugs such as bad/unsupported compiler flags being passed to meson, especially in complex meta build systems like buildroot which have a lot of intertwined packages.

> I've never seen it misused in that way, have you?

No, but the docs for `ld` very explicitly spelling out this `--just-symbols` behavior is supported by `ld` worries me enough that someone might be using this option that we really should catch it one way or another.